src/mbgl/programs/extrusion_texture_program.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <mbgl/programs/program.hpp>
+#include <mbgl/programs/attributes.hpp>
+#include <mbgl/programs/uniforms.hpp>
+#include <mbgl/shaders/extrusion_texture.hpp>
+#include <mbgl/style/layers/fill_extrusion_layer_properties.hpp>
+#include <mbgl/util/geometry.hpp>
+
+#include <string>
+
+namespace mbgl {
+
+struct ExtrusionTextureLayoutAttributes : gl::Attributes<
+attributes::a_pos>
+{};
+
+struct ExtrusionTextureUniforms : gl::Uniforms<
+uniforms::u_matrix,
+uniforms::u_world,
+uniforms::u_image,
+uniforms::u_opacity>
+{};
+
+class ExtrusionTextureProgram : public Program<
+shaders::extrusion_texture,
+gl::Triangle,
+ExtrusionTextureLayoutAttributes,
```

Comment:
If the type names aren't needed anywhere else, go ahead and inline `ExtrusionTextureLayoutAttributes` and `ExtrusionTextureUniforms`:

```C++
class ExtrusionTextureProgram : public Program<
    shaders::extrusion_texture
    gl::Triangle,
    gl::Attributes<
        attributes::a_pos>,
    gl::Uniforms<
        uniforms::u_matrix,
        uniforms::u_world,
        uniforms::u_image,
        uniforms::u_opacity>,
    style::PaintProperties<>>
```